Trinap/Sources/Presenter/Common/Reusable/TrinapSingleSelectionCalendarView.swift

```diff
@@ -0,0 +1,185 @@
+        }
+        .monthHeaderItemProvider { month in
+            return self.configureMonthHeaderItemProvider(month)
+        }
+    }
+    
+    private func configureDayItemProvider(day: Day, calendar: Calendar) -> CalendarItemModel<DayView> {
+        var invariantViewProperties = DayView.InvariantViewProperties.baseInteractive
+        
+        if self.possibleDate.contains(where: { calendar.component(.day, from: $0) == day.day }) {
+            if let selectedDate = self.selectedDate,
+                calendar.component(.day, from: selectedDate) == day.day {
+                invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 16)
+                invariantViewProperties.textColor = TrinapAsset.white.color
+                invariantViewProperties.backgroundShapeDrawingConfig.fillColor = TrinapAsset.primary.color
+            }
+        } else {
+            invariantViewProperties = DayView.InvariantViewProperties.baseNonInteractive
+            invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 16)
+            invariantViewProperties.textColor = TrinapAsset.disabled.color
+        }
+        
+        return DayView.calendarItemModel(
+            invariantViewProperties: invariantViewProperties,
+            viewModel: .init(
+                dayText: "\(day.day)",
+                accessibilityLabel: nil,
+                accessibilityHint: nil
+            )
+        )
+    }
+    
+    private func configureMonthHeaderItemProvider(_ month: Month) -> CalendarItemModel<MonthHeaderView> {
+        var invariantViewProperties = MonthHeaderView.InvariantViewProperties.base
+        
+        let trinapDoubleOffset = self.trinapOffset * 2
+        invariantViewProperties.edgeInsets = .init(
+            top: trinapDoubleOffset,
+            leading: trinapDoubleOffset,
+            bottom: 0,
+            trailing: 0
+        )
+        invariantViewProperties.font = TrinapFontFamily.Pretendard.bold.font(size: 20)
+        
+        return MonthHeaderView.calendarItemModel(
+            invariantViewProperties: invariantViewProperties,
+            viewModel: .init(
+                monthText: "\(month.year)년 \(month.month)월",
+                accessibilityLabel: nil
+            )
+        )
+    }
+    
+    private func filterPreviousDate(with possibleDate: [Date]) -> [Date] {
+        return possibleDate.filter { date in
+            let year = self.calendar.component(.year, from: date)
+            
+            return self.calendar.component(.year, from: Date()) == year
+        }
+        .filter { date in
+            let month = self.calendar.component(.month, from: date)
+            
+            return self.calendar.component(.month, from: Date()) == month
+        }
+        .filter { date in
+            let day = self.calendar.component(.day, from: date)
+            
+            return self.calendar.component(.day, from: Date()) <= day
+        }
```

Comment:
이 부분도 하나로 처리되면 좋을 것 같습니다!!

```suggestion
        return possibleDate.filter { date in
            let year = self.calendar.component(.year, from: date)
            let month = self.calendar.component(.month, from: date)
            let day = self.calendar.component(.day, from: date)
            
            return (
                self.calendar.component(.year, from: Date()) == year &&
                self.calendar.component(.month, from: Date()) == month &&
                self.calendar.component(.day, from: Date()) <= day &&
            )
        }
```

Reply:
이부분 또한 `dateComponents([.year, .month, .day], from: Date())` 사용해도 좋을 것 같습니다!

Reply:
수정.